### PR TITLE
Adding console log formatting article

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Take @maryamariyan's fork/branch, updating:
- Migrate code in snippets
- Add this to the TOC
- Editorial polish

Closes #21182

## Preview

✔️ [Console log formatting](https://review.docs.microsoft.com/en-us/dotnet/core/extensions/console-log-formatter?branch=pr-en-us-21189)

### CODE REVIEW

File: docs/core/extensions/console-log-formatter.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+| <xref:Microsoft.Extensions.Logging.Console.ConsoleFormatterNames.Simple?displayProperty=nameWithType> | <xref:Microsoft.Extensions.Logging.ConsoleLoggerExtensions.AddSimpleConsole%2A?displayProperty=nameWithType> |
+| <xref:Microsoft.Extensions.Logging.Console.ConsoleFormatterNames.Systemd?displayProperty=nameWithType> | <xref:Microsoft.Extensions.Logging.ConsoleLoggerExtensions.AddSystemdConsole%2A?displayProperty=nameWithType> |
+
+### Simple
+
+To use the `Simple` console formatter, register it with `AddSimpleConsole`:
+
+:::code language="csharp" source="snippets/logging/console-formatter-simple/Program.cs" highlight="11-16":::
+
+In the proceeding sample source code, the <xref:Microsoft.Extensions.Logging.Console.ConsoleFormatterNames.Simple?displayProperty=nameWithType> formatter was registered. It provides logs with the ability to not only wrap information such as time and log level in each log message, but also allows for ANSI color embedding and indentation of messages.
```
Comment (IEvangelist) [inline docs/core/extensions/console-log-formatter.md:41]:
```suggestion
In the preceding sample source code, the <xref:Microsoft.Extensions.Logging.Console.ConsoleFormatterNames.Simple?displayProperty=nameWithType> formatter was registered. It provides logs with the ability to not only wrap information such as time and log level in each log message, but also allows for ANSI color embedding and indentation of messages.
```

File: docs/core/extensions/console-log-formatter.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+Create an extension method to handle this for you:
+
+:::code language="csharp" source="snippets/logging/console-formatter-custom/ConsoleLoggerExtensions.cs" highlight="11-12":::
+
+The `CustomOptions` are defined as follows:
+
+:::code language="csharp" source="snippets/logging/console-formatter-custom/CustomOptions.cs":::
+
+In the proceeding code, the options are a subclass of <xref:Microsoft.Extensions.Logging.Console.ConsoleFormatterOptions>.
```
Comment (IEvangelist) [inline docs/core/extensions/console-log-formatter.md:110]:
```suggestion
In the preceding code, the options are a subclass of <xref:Microsoft.Extensions.Logging.Console.ConsoleFormatterOptions>.
```

File: docs/core/extensions/console-log-formatter.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+- Handles configuration:
+  - Uses a change token to synchronize updates, based on the [options pattern](options.md), and the [IOptionsMonitor](xref:Microsoft.Extensions.Options.IOptionsMonitor%601) interface
+
+:::code language="csharp" source="snippets/logging/console-formatter-custom/Program.cs" highlight="11-12":::
+
+Define a `CustomerFormatter` subclass of `ConsoleFormatter`:
+
+:::code language="csharp" source="snippets/logging/console-formatter-custom/CustomFormatter.cs" highlight="24-45":::
+
+The proceeding `CustomFormatter.Write<TState>` API dictates what text gets wrapped around each log message. A standard `ConsoleFormatter` should be able to wrap around scopes, time stamps, and severity level of logs at a minimum. Additionally, you can encode ANSI colors in the log messages, and provide desired indentations as well. The proceeding implementation for `CustomFormatter.Write<TState>` lacks these capabilities.
```
Comment (IEvangelist) [inline docs/core/extensions/console-log-formatter.md:124]:
```suggestion
The preceding `CustomFormatter.Write<TState>` API dictates what text gets wrapped around each log message. A standard `ConsoleFormatter` should be able to wrap around scopes, time stamps, and severity level of logs at a minimum. Additionally, you can encode ANSI colors in the log messages, and provide desired indentations as well. The implementation of the `CustomFormatter.Write<TState>` lacks these capabilities.
```

**Review by BillWagner — APPROVED** [review]
This LGTM @IEvangelist 

You can :shipit: when ready.

**Pull request merged**